doc/src/manual/metaprogramming.md

```diff
@@ -864,6 +864,47 @@ However, we don't do this for a good reason: wrapping the `expr` in a new scope
 also slightly changes the meaning of the expression (the scope of any variables in it),
 while we want `@time` to be usable with minimum impact on the wrapped code.
 
+### Macros and dispatch
+
+Macros can have multiple methods and dispatch based on their arguments, like ordinary functions:
```

Comment:
How about?

> Macros are *generic* (as well as functions¹) and can have multiple ...

¹ . or something like that, my English is not well enough to find the correct phrase.

Reply:
For me it is not clear, what "Macros are generic" means?

Reply:
*generic macros*, means that they can have multiple methods, just like *generic functions*, maybe it would help if this was mentioned also in `show`, just like it's done for functions, ie:

`@foo (generic macro with 1 method)`

But anyway that is precisely what the word generic means for functions, so it should be also used for macros IMO.

Reply:
How about? _Macros, just like functions, are *generic*, which means that they can also have multiple methods..._

Reply:
The meaning of generic is still vague to me. For example functions in say python (you have duck typing but no multiple dispatch there) are they generic too?

Reply:
No that is not what generic means, Python functions are not generic in this sense. Here is a wiki article:

* https://en.wikipedia.org/wiki/Generic_function

It is basically describing multiple dispatch also called multimethods.

The difference with Julia and the Common Lisp examples there, is that CL has both, normal functions and generic functions (with multiple dispatch, python has packages for this too). But in Julia **all** functions are by default generic.

Reply:
Here you can see again almost the same example of CL and below the example in Julia:

* https://en.wikipedia.org/wiki/Multiple_dispatch#Examples

Reply:
Thanks. Is "generic function" mostly Lisp jargon? Or is the term also used elsewhere?

Reply:
I believe so, it seems the term originated from Lisp:

* http://wiki.c2.com/?GenericFunction

Every description I get references CLOS and Common Lisp.

But if this term is ambiguous we should then explicitly say what it means in Julia land. When the manual was originally created it had lots of references from other languages without explaining what those where. Things like "julia has lisp like macros..." is very ambiguous for example.

Reply:
I tried to add an explanation what "generic function" means, but I did neither find a good place, nor a good formulation. I think explaining "generic function"  can be a separate PR.

Reply:
Defining "generic function" is a fairly tall order. Not that it's that hard to do, but it's a fairly fundamental concept in the language.